packages/good-design/src/core/web3/SwitchChainActionModal.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { FC, useMemo, useCallback, useEffect } from "react";
+import { useModal } from "../../hooks/useModal";
+import { Box, Text, useColorModeValue } from "native-base";
+import BasePressable from "../buttons/BasePressable";
+import { Title } from "../layout";
+import BackToSchool from "../../assets/images/backtoschool.png";
+import { Image } from "../images";
+import { openLink } from "@gooddollar/web3sdk-v2";
+
+export interface SwitchChainProps {
+  /**  during a active switch request, the modal pops up */
+  switching: boolean;
+}
+
+const SwitchChainActionModal: FC<SwitchChainProps> = ({ switching }: { switching: boolean }) => {
+  const { Modal: SwitchActionModal, showModal, hideModal } = useModal();
+  const textColor = useColorModeValue("goodGrey.500", "white");
+
+  const openNotionTab = useCallback(async () => {
+    const link = "https://www.notion.so/gooddollar/User-Guides-24dd615eb7804792a44057b96b40147d";
+    await openLink(link, "_blank");
+  }, []);
+
+  useEffect(() => {
+    switching ? showModal() : hideModal();
```

Comment:
ternary should be used for **evaluate** not for invoke

```suggestion
    const onSwitchingStateChanged = switching ? showModal : hideModal;
    
    onSwitchingStateChanged();
```

or you could use `if ... else`:

as one-liner
```suggestion
    if (switching) showModal(); else hideModal();    
```

or with code blocks (much better for readability)
```suggestion
    if (switching) {
      showModal();
    } else {
      hideModal();    
    }  
```